### ISSUE TITLE
[Mittorn's legacy launcher] Black bar on surface graphical bug (including flashlight lower fps and hologram lag spike)

### ISSUE THREAD

**Issue by TheKingFireS** [FWGS/xash3d-fwgs#1774]
To reproduce this bug, pick any maps (best case is hazard course map).

Looking at hologram will cause massive lag spike
sometime turn on flashlight will cause -15 fps.

``gl_vbo`` is hardcoded disabled.
``gl_dlightmode`` is disabled and enabled, still same effect.
switching to GLES2 will cause black screen (probably unrelated)
switching to GL4ES doesn't have black bar bug, but lower performance.
switching to Software doesn't have black bar bug and and it reduced hologram lag spike, but it's Software mode.
 
Device: Samsung Galaxy Fame Lite
OS: Android 4.1.2 with TouchWiz armv7 32bit
CPU: Broadcom BCM21654/G clocked 849MHz, 1xCortex-A9 @ 849MHz (yes, only 1 core)
GPU: Broadcom VideoCore IV HW, GLES 2.0
RAM: 512 MB
Engine: 0.20 ver 3178, commit e43015ec-dirty
(can't install latest version due to android 5 mininium and probably same bug "Can't find libSDL2.so" for Android 4.X)

Black bar (didn't happened on old engine):
![image](https://github.com/user-attachments/assets/538ba39e-3662-4cce-a787-87c1f81a1425)
Massive lag spike when looking at hologram entity (same happened on old engine):
![Screenshot_2007-04-08-05-30-04](https://github.com/user-attachments/assets/86ceb522-903e-4d03-88ad-ba6572068ad6)

**Comment by Vladislav4KZ** [comment]
@TheKingFireS To workaround the bug in the mainui with the buggy renderer selection (when GLES3Compat is actually the Software renderer) use the ``-ref gles3compat`` command line argument.

Does it have such issues?

**Comment by mittorn** [comment]
Is this gles1 renderer?

Seems to be some nanogl or driver bug.

Of course, you may try gles3compat renderer, it should work on any gles2 
hardware, but it may be even slower than gl4es on pure gles2 hardware 
(probably except drivers, supporting persistent/coherent buffer storages).
I made some dlight improvements in vbo-related branches later, but not 
published android builds. Rebasing to it might help

**Comment by TheKingFireS** [comment]
> Is this gles1 renderer?

Yes.

> Of course, you may try gles3compat renderer, it should work on any gles2 hardware, but it may be even slower than gl4es on pure gles2 hardware (probably except drivers, supporting persistent/coherent buffer storages).

Ok, I'll try.

**Comment by TheKingFireS** [comment]
> Of course, you may try gles3compat renderer, it should work on any gles2 hardware, but it may be even slower than gl4es on pure gles2 hardware (probably except drivers, supporting persistent/coherent buffer storages).

True, switching to gles3compat renderer, almost same as gles1/nanogl renderer (black bar bug, hologram lag spike and flashlight fps reducer)

**Comment by TheKingFireS** [comment]
Turnout, my Fame Lite phone don't really like rendering NPC models.

**Comment by TheKingFireS** [comment]
Hello, I juse realized I can use GLES capsviewer app to submit phone and driver info.
Here's result of this phone's driver:
https://opengles.gpuinfo.org/displayreport.php?id=7720
Hopefully this info will be useful info for you.
